components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java

```diff
@@ -80,7 +80,9 @@ public MonitoringFilter createMonitoringFilter() {
         switch (monitorFilterType) {
             default:
                 return new DataChangeFilter(
-                        DataChangeTrigger.StatusValueTimestamp, this.getDataChangeFilterDeadbandType(),
+                        this.dataChangeFilterTrigger != null
+                                ? this.dataChangeFilterTrigger : DataChangeTrigger.StatusValueTimestamp,

```

Comment:
Warning this is a behavior change, as far as I can see the current default value is [`DataChangeTrigger.StatusValue`](https://github.com/apache/camel/blob/c1a8be35640603c35160aba07d715c3cd3878af0/components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java#L41) whereas at runtime it used `DataChangeTrigger.StatusValueTimestamp`. Maybe should change the default value to `DataChangeTrigger.StatusValueTimestamp` and use `dataChangeFilterTrigger` here.

Reply:
Good point. I did not pay attenation to that the filter had a default different from what was actually applied here. 

I've updated the pullrequest to use the old effective default value (StatusValueTimestamp)

Reply:
For the sake of consistency with the rest of the code, don't check if it is `null` only call the corresponding getter